File: solutions/python/contains-duplicate.py
# id: 010
# title: Contains Duplicate
# approach: Hash Set
# time: O(n) | space: O(n)
# author: chron303

import sys

def main():
    data = sys.stdin.read().split()
    n = int(data[0])
    nums = list(map(int, data[1:n+1]))

    seen = set()
    for num in nums:
        if num in seen:
            print("true")
            return
        seen.add(num)

    print("false")

if __name__ == "__main__":
    main()

File: solutions/java/contains-duplicate.java
// id: 010
// title: Contains Duplicate
// approach: Hash Set
// time: O(n) | space: O(n)
// author: chron303

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine().trim());
        String[] input = br.readLine().trim().split(" ");

        Set<Integer> seen = new HashSet<>();
        for (int i = 0; i < n; i++) {
            int num = Integer.parseInt(input[i]);
            if (seen.contains(num)) {
                System.out.println("true");
                return;
            }
            seen.add(num);
        }

        System.out.println("false");
    }
}
